working/0546-patterns/patterns-feature-specification.md

```diff
@@ -1646,7 +1771,7 @@ that contains the pattern:
 *   **Switch statement case**: The guard clause and the statements of the
     subsequent non-empty case body.
 *   **Switch expression case**: The guard clause and the case expression.
-*   **Pattern-if statement**: The then statement.
+*   **If-case statement**: The guard clause and then statement.
 
 Multiple switch case patterns may share the same variable scope if their case
 bodies are empty:
```

Comment:
Nope. Just thought about this, and it *cannot* work.

```dart
late int Function() otther;
...
switch([1, 2]) {
  case [final int a, final int b] when (other = ()=>a) == 0:
  case [final int b, final int a]:
     print(a + b + other(); // 2 + 1 + 1
```

The scope of each `casehead` must be independent, because it can bind *final* variables to values even if then refuted by a `when` clause. The same final variable cannot be initialized more than once. 
And since a `when` clause is an expression, it can capture the final variable, and must see the correct value.

So each `case` pattern introduces its own bindings.

We then have to make *one* of those sets of bindings the current binding for the body, when we execute/evaluate that body. That's a little weird, but not much weirder than what we do for `for(var x = 1;x < 10;x++)` iterations. (Which isn't really saying much. That *is* weird.)

See more above.